vonk/reference/programming_api/repository.rst

```diff
@@ -0,0 +1,212 @@
+   #. Call IArgument.Handled() to update its status. There is also .Warning() and .Error() when something is wrong with the argument. If you simply don't support the argument, you can leave the status to 'Unhandled'. 
+   #. Pay special attention to the ``_count`` and ``_skip`` arguments for proper paging.
+
+#. 'AND' all the arguments together, e.g. forming a database query or complete querystring.
+
+#. Issue the query to your repository and await the results (await used intentionally: this should be done asynchronously).
+
+#. For each of the resulting records or objects: map them to matching FHIR resources, either by: 
+
+   #. Creating POCO's:
+   
+      .. code-block:: csharp
+         
+         var result = new Patient() { /*fill in from the source object*/ };
+         return result.ToIResource(); //InformationModel implied by the assembly of class Patient
+         
+   #. or by crafting SourceNodes:
+   
+      .. code-block:: csharp
+      
+         var result = SourceNode.Resource("Patient", SourceNode.Valued("id", /* id from source */), SourceNode.Node("meta", SourceNode.Valued("versionId", "v1"), ....), ...))
+         return result.ToIResource(VonkConstants.Model.FhirR3 /* or FhirR4 */);
+
+#. Combine the mapped resources into a SearchResult:
+
+   .. code-block:: csharp
+
+      return new SearchResult(resources, pagesize, totalCount, skip);
+
+   * ``pagesize``: should be the value of the _count argument, unless you changed it for some reason.
+   * ``totalCount``: total number of results, if there are more than you are returning right now.
+   * ``skip``: number of results skipped in this set (if you are serving page x of y).
+   
+For a Facade on a relational database we provide a starting point with ``Vonk.Facade.Relational.SearchRepository``. Follow the exercise in :ref:`facadestart` to see how that is done.
+
+.. _vonk_reference_api_ichangerepository:
+
+IResourceChangeRepository
+-------------------------
+
+``IResourceChangeRepository`` defines methods to change resources in the repository:
+
+.. code-block:: csharp
+
+   public interface IResourceChangeRepository
+   {
+      Task<IResource> Create(IResource input);
+      Task<IResource> Update(ResourceKey original, IResource update);
+      Task<IResource> Delete(ResourceKey toDelete, string informationModel);
+      string NewId(string resourceType);
+      string NewVersion(string resourceType, string resourceId);
+   }
+
+``ResourceKey`` is a simple struct to identify a resource by Type, Id and optionally VersionId.
+
+Using IResourceChangeRepository
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You should hardly ever need to use the ``IResourceChangeRepository``. It is used by the :ref:`Create <vonk_plugins_create>`, :ref:`Update <vonk_plugins_update>`, :ref:`Delete <vonk_plugins_delete>` and the conditional variations thereof.
+
+Should you need to use it, the methods are fairly straightforward.
+
+:method: Create
```

Comment:
Somehow this part is formatted wider than the page, so it looks odd. Not something you need to focus on right now, but perhaps we can revisit and do a layout check after DevDays.

Reply:
Yes, weird. I actually want this type of documentation generated from the source code comments. Afaik that is not easily done with rst, but can be done with DocFX. I'd rather spend my time on that :-)